### PR TITLE
bugfix: add auth flag for cluster scale

### LINKED ISSUE #1469: Sealer subcommand "join" doesn't support --passwd, --pk, --env, which is supported in subcommand "run".
image

image

Sealer subcommand "join" doesn't support --passwd, --pk, --env, which is supported in subcommand "run". 

This design requires that the configuration of the node I join must be consistent with the original node, which is unreasonable

#### What you expected to happen:
I expect when i use sealer join, i can set ssh config, env for new nodes.

#### Environment:

- sealer version (use `sealer version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe what this PR does / why we need it

1. add ssh auth flag and env for sealer join.
2. add env flag for sealer delete.
3. split cluster v1 and cluster v2 save logic at Clusterfile saving stage.
4. avoid adding empty struct filed  of Clusterfile.

### Does this pull request fix one issue?

fixes #1469 

### Describe how to verify it

test success  on my side:

all cluster node use the same password:

```shell
sealer run kubernetes:v1.21.12 -m 172.16.0.200 -p password
sealer join kubernetes:v1.21.12 -n 172.16.0.201
sealer delete -n 172.16.0.201
```

```shell
sealer run kubernetes:v1.21.12 # local host as master0, no need to set password
sealer join kubernetes:v1.21.12 -n 172.16.0.201 -p password123 # if join another node, need to set password.
sealer delete -n 172.16.0.201
```

all cluster node use the different  password:

```shell
sealer run kubernetes:v1.21.12 -m 172.16.0.200 -p password
sealer join kubernetes:v1.21.12 -n 172.16.0.201 -p password123
sealer delete -n 172.16.0.201 
```

### CODE REVIEW

File: cmd/sealer/cmd/delete.go [hunk 1]
```diff
@@ -96,6 +96,7 @@ func init() {
 	deleteCmd.Flags().StringVarP(&deleteArgs.Nodes, "nodes", "n", "", "reduce Count or IPList to nodes")
 	deleteCmd.Flags().StringVarP(&deleteClusterFile, "Clusterfile", "f", "", "delete a kubernetes cluster with Clusterfile Annotations")
 	deleteCmd.Flags().StringVarP(&deleteClusterName, "cluster", "c", "", "delete a kubernetes cluster with cluster name")
+	deleteCmd.Flags().StringSliceVarP(&deleteArgs.CustomEnv, "env", "e", []string{}, "set custom environment variables")
```
Comment (allencloud) [inline cmd/sealer/cmd/delete.go:99]: set or delete?

Comment (kakaZhou719) [reply cmd/sealer/cmd/delete.go:99]: its `set` , thats because in some case user want to set custom Env params for supporting execute clean scripts  .

**Review by bxy4543 — APPROVED** [review]

**Comment by VinceCui** [comment]
LGTM

**Pull request merged**